### PR TITLE
[UDT PR 6/N] Add default controller to async controller registry

### PR DESCRIPTION
**Pull request by rynowak**
# Description

This change adds a "default" controller capability to the controller registry used by the async worker. This is needed for the dynamic-rp. Because the dynamic-rp is extensible, it processes many different types of operations, it's not possible for us to statically register all of these. Instead, the dynamic-rp will use this new default capability to register a single controller with dynamic functionality.

## Type of change

- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius (issue link optional).

## Contributor checklist
Please verify that the PR meets the following requirements, where applicable:

- [ ] An overview of proposed schema changes is included in a linked GitHub issue.
- [ ] A design document PR is created in the [design-notes repository](https://github.com/radius-project/design-notes/), if new APIs are being introduced.
- [ ] If applicable, design document has been reviewed and approved by Radius maintainers/approvers.
- [ ] A PR for the [samples repository](https://github.com/radius-project/samples) is created, if existing samples are affected by the changes in this PR.
- [ ] A PR for the [documentation repository](https://github.com/radius-project/docs) is created, if the changes in this PR affect the documentation or any user facing updates are made.
- [ ] A PR for the [recipes repository](https://github.com/radius-project/recipes) is created, if existing recipes are affected by the changes in this PR.

### CODE REVIEW

File: pkg/armrpc/asyncoperation/worker/registry.go [hunk 1]
```diff
@@ -65,14 +70,46 @@ func (h *ControllerRegistry) Register(ctx context.Context, resourceType string,
...
 	}
 
-	return nil
+	return h.createDefault(ctx, operationType)
+}
+
+func (h *ControllerRegistry) createDefault(ctx context.Context, operationType v1.OperationType) (ctrl.Controller, error) {
+	if h.defaultFactory == nil {
+		return nil, nil
+	}
```
Comment (ytimocin) [inline pkg/armrpc/asyncoperation/worker/registry.go:101]: Isn't this function more like **getDefault**? If **defaultFactory** is nil, then we return nil for the controller. So, to me, it is more like **getDefault** that is called from the **Get** function if no controller is registered.

Comment (rynowak) [reply pkg/armrpc/asyncoperation/worker/registry.go:101]: Is this feedback about the naming or behavior? I don't have a strong opinion about the name 😆

Comment (ytimocin) [reply pkg/armrpc/asyncoperation/worker/registry.go:101]: It is about the name of the function not the behavior. It seems like it doesn't create the default but get the existing default.

**Comment by rynowak** [comment]
Functional test failure appeared to be contention in azure on a storage account. If it happens again I'll disable it.

**Comment by rynowak** [comment]
@ytimocin - updated

**Review by ytimocin — APPROVED** [review]

**Pull request merged**